tests/api/test_result.py

```diff
@@ -16,30 +16,99 @@
 """
 import numpy as np
 import pytest
+from strawberryfields.backends.states import BaseGaussianState
 
 from strawberryfields.result import Result
 
 pytestmark = pytest.mark.api
 
 
+raw_results = {
+    "output": [np.ones((2, 3, 4, 5))],
+    0: [1, 2, 3],
+    1: [4, 5],
+    "other_data": [1, 0, 1, 2],
+    "string": "metadata",
+}
+
+
 class TestResult:
     """Tests for the ``Result`` class."""
 
+    def test_samples(self):
+        """Test that ``samples`` is correctly returned."""
+        result = Result(raw_results)
+        assert result.samples is not None
+        assert np.equal(result.samples, np.ones((2, 3, 4, 5))).all()
+
+    def test_raw(self):
+        """Test that raw results are correctly returned."""
+        result = Result(raw_results)
+        assert result.raw == raw_results
+
+    def test_samples_dict(self):
+        """Test that ``samples_dict`` is correctly returned."""
+        result = Result(raw_results)
+        assert result.samples_dict == {0: [1, 2, 3], 1: [4, 5]}
```

Comment:
I just realized that SF v0.20.0 may not be handling the samples dictionary correctly - the `Result` class should be checking for _strings_ in the raw result which resemble integers.

Reply:
The usage of integers as keys was deliberate here. See my other comment above.

Reply:
Let's continue this discussion in the above comment!